dags/deletes.py

```diff
@@ -708,56 +672,60 @@ def wait_for_delete_mutations_in_all_hosts(
     return pending_deletes_dict
 
 
+@dataclass
+class VerifiedDeletionResources:
+    pending_deletions_dictionary: PendingDeletesDictionary
+    adhoc_event_deletes_dictionary: AdhocEventDeletesDictionary
+
+
+@dagster.op
+def mark_deletions_verified(
+    cluster: dagster.ResourceParam[ClickhouseCluster],
+    pending_deletions_dictionary: PendingDeletesDictionary,
+    adhoc_event_deletes_dictionary: AdhocEventDeletesDictionary,
+) -> VerifiedDeletionResources:
+    now = timezone.now()
+    deletion_ids = [
+        id
+        for (id,) in cluster.any_host_by_role(
+            Query(f"SELECT id FROM {pending_deletions_dictionary.source.qualified_name}"), node_role=NodeRole.DATA
+        ).result()
+    ]
+    for chunk in chunked(deletion_ids, n=10000):
+        AsyncDeletion.objects.filter(id__in=chunk).update(delete_verified_at=now)
```

Comment:
This is the important bit, now we only mark rows that were present in the table at the start of the run and used for populating the dictionary as verified https://posthog.slack.com/archives/C076R4753Q8/p1751058786220829?thread_ts=1750870642.615099&cid=C076R4753Q8